default.nix

```diff
@@ -1,138 +1,35 @@
-            pkgs.zstd.dev
-            pkgs.lz4.dev
-            pkgs.perl
-            pkgs.patch
-          ];
-        });
-
-      ethereum = dontCheck (self.callCabal2nix "ethereum" (pkgs.fetchFromGitHub {
-        owner = "kadena-io";
-        repo = "kadena-ethereum-bridge";
-        rev = "10f21e96af1dce4f13e261be9dfad8c28cd299f7";
-        sha256 = "1vab2m67ign6x77k1sjfjmv9sbrrl5sl2pl07rw1fw8bjqnp5vqk";
-      }) {});
-
-      resource-pool = self.callHackageDirect {
-        pkg = "resource-pool";
-        ver = "0.4.0.0";
-        sha256 = "0zlnizx0igwmvpx43mcnk0h58v9lakhwr4g9csy1vj74p7a4hxaz";
-      } {};
-
-      direct-sqlite = self.callHackageDirect {
-        pkg = "direct-sqlite";
-        ver = "2.3.27";
-        sha256 = "0w8wj3210h08qlws40qhidkscgsil3635zk83kdlj929rbd8khip";
-      } {};
-
-      yet-another-logger = self.callHackageDirect {
-        pkg = "yet-another-logger";
-        ver = "0.4.1";
-        sha256 = "1qb0ns764sb5az8z1dn7pflizi8ni8qivbhx79sj9kfaa68hyhsl";
-      } {};
-
-      hashes = self.callHackageDirect {
-        pkg = "hashes";
-        ver = "0.2.2.1";
-        sha256 = "1039smdvx03j52md2vv1lvkllm47rf0zi9bqmnh1lk1zhcv0sag3";
-      } {};
-
-      pact = appendConfigureFlag super.pact "-f-build-tool";
-
-      autodocodec    = unmarkBroken super.autodocodec;
-      hashable       = doJailbreak super.hashable;
-      ixset-typed    = unmarkBroken super.ixset-typed;
-      rebase         = doJailbreak super.rebase;
-      token-bucket   = unmarkBroken super.token-bucket;
-      validity-aeson = unmarkBroken super.validity-aeson;
-
-      # Cuckoo tests fail due to a missing symbol
-      cuckoo        = dontCheck super.cuckoo;
-
-      # These tests pull in unnecessary dependencies
-      http2         = dontCheck super.http2;
-      prettyprinter = dontCheck super.prettyprinter;
-      aeson         = dontCheck super.aeson;
-      generic-data  = dontCheck super.generic-data;
-  };
-
-  source-overrides = {
-    chainweb-storage = nix-thunk.thunkSource ./dep/chainweb-storage;
-    pact             = nix-thunk.thunkSource ./dep/pact;
-
-    OneTuple                    = "0.3";
-    aeson                       = "1.5.6.0";
-    ansi-terminal               = "0.11.3";
-    prettyprinter-ansi-terminal = "1.1.2";
-    time-compat                 = "1.9.5";
-    trifecta                    = "2.1.1";
-    unordered-containers        = "0.2.15.0";
-
-    # These are required in order to not break payload validation
-    base16-bytestring = "0.1.1.7";
-    prettyprinter     = "1.6.0";
-    hashable          = "1.3.0.0";
-    base64-bytestring = "1.0.0.3";
-  };
-
-  modifier = drv: pkgs.haskell.lib.overrideCabal drv (attrs: {
-    buildTools = (attrs.buildTools or []) ++ [
-      pkgs.zlib
-      pkgs.haskell.packages.${compiler}.cabal-install
-    ];
-  });
-
-  inherit returnShellEnv;
+{ pkgs ? pkgsDef
+, compiler ? "ghc8107"
+, ...
+}:
+let chainweb-node = pkgs.haskell-nix.project' {
+	    src = ./.;
```

Comment:
Looks like this gets rid of nix-gitignore.

Flake input:
```
nix-gitignore.url = github:hercules-ci/gitignore.nix";
```

Reply:
Arguably we would want a shared `src` attribute between both appearances in `default.nix`, so they have consistent ignores.

Reply:
Thanks for pointing out this issue @chessai! I think using `hercules-ci/gitignore.nix` is a good suggestion, but AFAIK, that only filters out the files that are ignored by git, but it will accept new files that are neither ignored nor tracked by git. I think it's a reasonable thing that it does, but unfortunately that behavior is not the same as how Nix flakes behave. A Nix flake will first be copied to the Nix store and it will get built there. While the Nix flake gets copied, only files that are tracked by git will be copied, but if there are any changes they will be copied with those changes.

Instead of going forward with `hercules-ci/gitignore.nix`, I've decided to keep `default.nix` compatible with `flake.nix` by using `src = self.outPath`. Here's a demonstration of how this makes `default.nix` equivalent to `flake.nix`:
```
$ nix build --print-out-paths
/nix/store/8q1h87dpbv595ik2r5sw6rz29v836qq0-chainweb-exe-chainweb-node-2.18.1

$ nix-build 
/nix/store/8q1h87dpbv595ik2r5sw6rz29v836qq0-chainweb-exe-chainweb-node-2.18.1

$ touch somenewfile
$ nix-build 
/nix/store/8q1h87dpbv595ik2r5sw6rz29v836qq0-chainweb-exe-chainweb-node-2.18.1
$ nix build --print-out-paths
/nix/store/8q1h87dpbv595ik2r5sw6rz29v836qq0-chainweb-exe-chainweb-node-2.18.1

$ echo changing tracked file >> README.md 
$ nix build --print-out-paths
warning: Git tree '.../chainweb-node' is dirty
/nix/store/rlxyjmwqalljkzg088dzwy0kd8bbaqnl-chainweb-exe-chainweb-node-2.18.1
$ nix-build 
warning: Git tree '.../chainweb-node' is dirty
/nix/store/rlxyjmwqalljkzg088dzwy0kd8bbaqnl-chainweb-exe-chainweb-node-2.18.1
```

Reply:
> Arguably we would want a shared `src` attribute between both appearances in `default.nix`, so they have consistent ignores.

Seems like this comment isn't applicable anymore with my latest changes.